nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/NuxeoOAuth2Servlet.java

```diff
@@ -340,6 +343,38 @@ protected void doPostToken(HttpServletRequest request, HttpServletResponse respo
             } else {
                 handleTokenResponse(refreshed, response);
             }
+        } else if (JWT_BEARER_GRANT_TYPE.equals(grantType)) {
+            String jwtToken = tokenRequest.getAssertion();
+            Map<String, Object> claims = Framework.getService(JWTService.class).verifyToken(jwtToken);
+            OAuth2Error error = null;
+
+            String clientId = tokenRequest.getClientId();
+            String clientSecret = tokenRequest.getClientSecret();
+
+            if (StringUtils.isBlank(clientId)) {
+                error = OAuth2Error.invalidRequest("Empty client id");
+            } else if (!clientService.hasClient(clientId)) {
+                error = OAuth2Error.invalidClient(String.format("Invalid client : %s", clientId));
+            } else if (!clientService.isValidClient(clientId, clientSecret)) {
+                error = OAuth2Error.invalidClient(
+                        String.format("Disabled client : %s or invalid client secret", clientId));
+            }
+
+            if (error != null) {
+                handleJsonError(error, response);
+                return;
+            }
+
+            String username = (String) claims.get(CLAIM_SUBJECT);
+            NuxeoOAuth2Token token = tokenStore.getToken(clientId, username);
+            if (token == null) {
+                final NuxeoOAuth2Token newToken = new NuxeoOAuth2Token(ACCESS_TOKEN_EXPIRATION_TIME, clientId);
+                TransactionHelper.runInTransaction(() -> tokenStore.store(username, newToken));
+                token = newToken;
+            }
```

Comment:
Shouldn't we need this part:
```
            } else if (token.isExpired()) {
                final String refreshToken = token.getRefreshToken();
                token = TransactionHelper.runInTransaction(() -> tokenStore.refresh(refreshToken, clientId));
            }
```
for case the token was already issued a long time ago.